parts/linux/cloud-init/artifacts/cgroup-stats.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+set -o errexit
+
+EVENTS_LOGGING_DIR=/var/log/azure/Microsoft.Azure.Extensions.CustomScript/events/
+EVENTS_FILE_NAME=$(date +%s%3N)
+CSE_STARTTIME=$(date)
+CSE_STARTTIME_FORMATTED=$(date +"%F %T.%3N")
+CSE_ENDTIME_FORMATTED=$(date +"%F %T.%3N")
+NODEPOOL_NAME=$(curl -s -H Metadata:true --noproxy "*" "http://169.254.169.254/metadata/instance?api-version=2021-02-01" | jq -r '.compute.osProfile.computerName')
+VM_TYPE=$(kubectl describe node $NODEPOOL_NAME --kubeconfig var/lib/kubelet/kubeconfig | grep "beta.kubernetes.io/instance-type=" | cut -d '=' -f 2)
+
+memory_string=$( jq -n \
```

Comment:
we're assuming cgroupv2 here but this may still run on v1 nodes (mariner in particular). 

we should either handle v1, use `memory.usage_in_bytes` or `memory.stat`, and distinguish here, or we should distinguish here and bail out early. 

`stat -fc %T /sys/fs/cgroup`

https://kubernetes.io/docs/concepts/architecture/cgroups/#check-cgroup-version

Reply:
if we handle v1 vs v2, can you emit the version as well. it would be implied by lack of PSI but nice to have explicitly.

Reply:
Technically if I used memory.stat wouldn't that work for both v1 and v2. Its just the pressure metrics that v1 doesn't have.

Reply:
yep, but the same values have different names.

cgroupv2 memory.stat.file = cgroupv1 memory.stat.cache = NR_FILE_PAGES
cgroupv2 memory.stat.anon = cgroupv1 memory.stat.rss = NR_ANON_MAPPED
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/mm/memcontrol.c#n4058
https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/mm/memcontrol.c#n1495